### PR TITLE
One Way Platform and Actor Spawner Fix

### PR DESCRIPTION
**Pull request by poec987**
My One Way Platform actor that supports movement and direction changing (180 degree turn for now...) via Event ID. Also has a fix to the actor spawner to prevent it from "deactivating" actors that are not it's child when it's child is deleted.

Kinda wish I can make a separate PR for the actor spawner fix, but GitHub is weird..

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Yeah, past me is stupid. I think I figured out how to make a separate pr.

### CODE REVIEW

File: source/actors/onewayplatform.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    , startrotation()
+    , endrotation()
+    , movementHandler()
+{ }
+
+u32 OneWayPlatform::onCreate() {
+
+    platform.create((settings1 >> 0x18 & 0xF), (settings1 >> 0x1c & 0xF));
+
+    if (platform.width < 2) platform.width = 2;
```
Comment (Luminyx1) [inline source/actors/onewayplatform.cpp:47]: Do if statements like this for clarity

if (x) {
   //...
}

File: source/actors/onewayplatform.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    platform.position.x = this->position.x;
```
Comment (Luminyx1) [inline source/actors/onewayplatform.cpp:49]: These 3 lines can be condensed into platform.position = this->position;

File: source/actors/onewayplatform.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+u32 OneWayPlatform::onExecute() {
+
+    u32 direction = settings1 >> 0x14 & 0xF;
+
+    this->movementHandler.execute();
+
+    this->position = this->movementHandler.position;
+
+    if(EventMgr::instance()->isActive(this->eventID1 - 1)) {
```
Comment (Luminyx1) [inline source/actors/onewayplatform.cpp:104]: Space after `if` please

**Review by Luminyx1 — APPROVED** [review]

File: source/actors/onewayplatform.cpp (hunk 3 above)
Comment (poec987) [reply source/actors/onewayplatform.cpp:104]: I can never unsee this 💀

**Pull request merged**